src/editor/sidebar/index.js

```diff
@@ -2,20 +2,44 @@
  * WordPress dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
-import { ExternalLink, PanelRow, ToggleControl } from '@wordpress/components';
+import {
+	BaseControl,
+	DateTimePicker,
+	ExternalLink,
+	PanelRow,
+	ToggleControl,
+} from '@wordpress/components';
 import { compose } from '@wordpress/compose';
 import { withDispatch, withSelect } from '@wordpress/data';
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { useEffect, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { isListing } from '../utils';
 import './style.scss';
 
-const SidebarComponent = ( { meta, updateMetaValue } ) => {
-	const { post_type_label: postTypeLabel, post_types: postTypes } = window.newspack_listings_data;
-	const { newspack_listings_hide_author, newspack_listings_hide_publish_date } = meta;
+const SidebarComponent = ( { createNotice, meta, publishDate, updateMetaValue } ) => {
+	const [ initialExpirationDate, setInitialExpirationDate ] = useState( null );
+	const {
+		is_listing_customer: isListingCustomer = false,
+		post_type_label: postTypeLabel,
+		post_types: postTypes,
+		self_serve_listing_expiration: expirationPeriod,
+	} = window.newspack_listings_data;
+	const {
+		newspack_listings_hide_author: hideAuthor,
+		newspack_listings_hide_publish_date: hidePublishDate,
+		newspack_listings_expiration_date: expirationDate,
+	} = meta;
+
+	// Get the last saved expiration date, if any.
+	useEffect( () => {
+		if ( expirationDate ) {
+			setInitialExpirationDate( expirationDate );
+		}
+	}, [] );
```

Comment:
Good call—updated in ea0c529.

Reply:
With these changes, isn't `initialExpirationDate === expirationDate`?

Reply:
It does, at first—but we want to maintain `initialExpiraitonDate` because customers shouldn't be able to set a post's expiration date beyond either a.) 30 days after purchase date, if there's no set expiration date, or b.) the set expiration date, if it occurs after the 30 day period (this condition would happen if the customer has extended the purchase for that post).